chapters/annotations.tex

```diff
@@ -1808,51 +1896,105 @@ \subsubsection{Conversion Rules}\label{conversion-rules}
 This is useful if there is no possibility to convert a specific parameter (or other element), especially if it rarely modified.  If the parameter had no impact on the model it can be removed using \lstinline!convertModifiers!, see \cref{convertmodifiers}.
 \end{nonnormative}
 
+
 \subsection{Versions in the File System}\label{mapping-of-versions-to-file-system}\label{versions-in-the-File-System}
 
-A top-level class, \lstinline!IDENT!, with version \lstinline!VERSION-NUMBER! can be stored in
-one of the following ways in a directory given in the \lstinline!MODELICAPATH!:
+The top-level class $\mathit{packageName}$ with version $\mathit{packageVersion}$ can be stored in one of the following ways in a directory given in the \lstinline!MODELICAPATH! (\cref{the-modelica-library-path-modelicapath}):
 \begin{itemize}
 \item
-  The file \lstinline!IDENT ".mo"!\\
+  The file $\mathit{packageName}$\filename{.mo}\\
   Example: \filename{Modelica.mo}
 \item
-  The file \lstinline!IDENT " " VERSION-NUMBER ".mo"!\\
+  The file $\mathit{packageName}$\textvisiblespace$\mathit{packageVersion}$\filename{.mo}\\
```

Comment:
\textvisiblespace looks a lot like an underscore in the pdf, depending on the zoom. It is not really clear that it should be a space and not something else
![image](https://user-images.githubusercontent.com/20460714/161122409-9bf26746-af7f-4ea6-9cda-7a45f8b00038.png)

Reply:
I'm not sure what better alternatives there are.  Can't the example be considered sufficient clarification; when you see the example, don't you realize what that little underscore-looking thing really means?

Reply:
Also, an important feature of the visible space is that it will also be there regardless of where line is wrapped – in general of special importance for readers of HTML on small screens (in theory we could always proof read the PDF to make sure lines are not wrapped at bad places…), although this case is too near the beginning of the line to really be at risk.

How about adding a comment for extra clarity?
```suggestion
  The file $\mathit{packageName}$\textvisiblespace$\mathit{packageVersion}$\filename{.mo} (with one space character separating $\mathit{packageName}$ from $\mathit{packageVersion}$)\\
```

Reply:
The only reason I know it's supposed to be a space and not an underscore (and a weird one at that) is because I look at the LaTeX source. So, my first reaction while looking at the pdf only was that there was a typo in the example and the underscore was missing. Actually, that's not true, my first reaction was: I had no idea underscore is supposed to separate the name and the version. I have to say that I find the grammar-style representation that was before this change, much clearer and concise. Since, we are not getting rid of the grammar-style anyways, at least for now, maybe we should just leave it as it was.

Reply:
I'd like to avoid that for several reasons.  The obvious one is that our overall goal is to avoid use of grammar whenever possible.  Another one is that `VERSION-NUMBER` was actually never defined, so we'd have to introduce even more grammar to give this a meaning.  The third is that we should technically use the lexing language (like when `IDENT` is defined) and not grammar in order to be able to express that there's no room for white-space or comments between the tokens.  If we'd like to avoid overuse of grammar, it should be even more important to avoid overuse of lexing.

Reply:
By the way, I don't think we need to be afraid of ending up adding comments like the one I sugest all over the document; the need to express separation by exactly one space character is unique to this section as far as I know, and therefore some kind of comment might actually be a good idea no matter what the formal specification looks like.

Reply:
I have been waiting to see if anybody else would comment, but nobody has volunteered so far. @HansOlsson, do you have an opinion?

Reply:
> Another one is that `VERSION-NUMBER` was actually never defined, so we'd have to introduce even more grammar to give this a meaning. 

I thought the definition of `VERSION-NUMBER` is given in [18.8.1 Version Numbering](https://specification.modelica.org/master/annotations.html#version-numbering). But I see that it never actually uses the term `VERSION-NUMBER`. Should it? It is used a lot in 18.8.2

Reply:
> I thought the definition of `VERSION-NUMBER` is given in [18.8.1 Version Numbering](https://specification.modelica.org/master/annotations.html#version-numbering). But I see that it never actually uses the term `VERSION-NUMBER`. Should it? It is used a lot in 18.8.2

As of this PR, it is not.  However, there is now a definition of the production rule _package-version_. Maybe it could be a solution to point out in the text that `\mathit{packageVersion}` needs to match _package-version_, along with a cross reference to the definition?

Reply:
I went ahead and implemented that idea, and it turned out that as _package-version_ was defined to be a package version string literal, I had to actually introduce _PACKAGE-VERSION_ for the contents of such as string, which is what the `\mathit{packageVersion}` has to match.

Reply:
That looks good. I'll keep it unresolved until the next week (it's a holiday here tomorrow and maybe holiday on Monday in other countries) to see if anybody else wants to chime in about the use of `\textvisiblespace`.

Reply:
Ok, there were no other comments. So I guess, it's OK to keep `\textvisiblespace`. The suggestion [above with the clarifying comment](https://github.com/modelica/ModelicaSpecification/pull/2957/files#r841122964), would be good. Sorry, I thought it was already committed. But I see now that it's only in the comments so far.

Reply:
Done.

Reply:
Please mark as resolved if OK.